ui/app/AppLayouts/Chat/components/PinnedMessagesPopup.qml

```diff
@@ -2,9 +2,32 @@ import QtQuick 2.13
 import "../../../../imports"
 import "../../../../shared"
 import "../../../../shared/status"
+import "../data"
 import "../ChatColumn"
 
 ModalPopup {
+    property bool userCanPin: {
+        switch (chatsModel.channelView.activeChannel.chatType) {
+        case Constants.chatTypePublic: return false
+        case Constants.chatTypeStatusUpdate: return false
+        case Constants.chatTypeOneToOne: return true
+        case Constants.chatTypePrivateGroupChat: return chatsModel.channelView.activeChannel.isAdmin(profileModel.profile.pubKey)
+        case Constants.chatTypeCommunity: return chatsModel.communities.activeCommunity.admin
+        default: return false
+        }
+    }
+    property bool canStillPin: {
+        if (!userCanPin) {
+            return false
+        }
+        const nbPinnedMessages = chatsModel.messageView.pinnedMessagesList.count
+
+        return nbPinnedMessages < Constants.maxNumberOfPins
```

Comment:
I'd assume there is no way to reuse previous logic?

Reply:
It would be annoying, because they are very different components. I could move the logic to Nim, but maybe it's overkill?

Reply:
Probably yes, it can stay that way